### PR TITLE
Upstream ac_fixed and hls_float Intel extensions

### PR DESCRIPTION
**Pull request by MrSidims**

### CODE REVIEW

**Comment by MrSidims** [comment]
@bashbaug @mkinsner could you please take a look? Additional enums for QuantizationModes, OverflowModes etc are currently missing, since I believe some work has to be done in the spec first, see: https://github.com/intel/llvm/pull/1934/files#r520667596

File: include/spirv/unified1/spirv.core.grammar.json
Comment (dneto0) [inline include/spirv/unified1/spirv.core.grammar.json:6716]: I recommend that things that are not capabilities should normally not have an extensions list.
Enable them by a capabilities list, and then have the capabilities enabled by the extensions list.
That makes it less confusing because then there is no question about whether it's (capabilties *and* extensions) vs. (capabilities *or* extensions).

**Review by dneto0 — APPROVED** [review]

**Comment by MrSidims** [comment]
I'll rebase it in a moment, thanks for merging another one.

**Review by bashbaug — APPROVED** [review]
I didn't do a comprehensive review for each instruction and its operands, but the structure of the changes looks correct, and the new instructions are tagged with `@exclude` so they shouldn't affect the SPIR-V spec generation.

For reference, the spec PR for these extensions is currently here: https://github.com/intel/llvm/pull/1934

Needs a rebase before merging.

File: include/spirv/unified1/spirv.core.grammar.json
Comment (dneto0) [inline include/spirv/unified1/spirv.core.grammar.json:6802]: I recommend using the "extensions" list only on the newly introduced capability enum;   the rest of the features are enabled by the capability only.
That removes doubt about how whether you need both the capability *and* the extension.

Comment (MrSidims) [reply include/spirv/unified1/spirv.core.grammar.json]: Done, thanks!

**Comment by MrSidims** [comment]
@dneto0 hi, what is remaining to do before the merge?

**Comment by svenvh** [comment]
@MrSidims there is a .swp file in this PR, you probably want to take that out before merging.

Other than that, it would be great to see this PR merged, to reduce the divergence of the SPIRV-LLVM-Translator's spirv.hpp.

**Comment by MrSidims** [comment]
> @MrSidims there is a .swp file in this PR, you probably want to take that out before merging.
> 
> Other than that, it would be great to see this PR merged, to reduce the divergence of the SPIRV-LLVM-Translator's spirv.hpp.

Oops, thanks!

File: include/spirv/unified1/spirv.core.grammar.json
Comment (mlychkov) [inline include/spirv/unified1/spirv.core.grammar.json:6885]: Please, take into account the changes in API of OpArbitraryFloatCastFromIntINTEL and OpArbitraryFloatCastToIntINTEL instructions (https://github.com/intel/llvm/pull/1934/files#diff-a253d8397fe69443a09260a0c2433ce7236b7ef23e90af92cb460100f96a20d2R1120).

Comment (AGindinson) [reply include/spirv/unified1/spirv.core.grammar.json:6887]: @mlychkov, done, thanks.

**Comment by AGindinson** [comment]
@svenvh, the remaining comment has been addressed. Should I rebase/squash the commits before the merge can be done?

**Comment by svenvh** [comment]
> @svenvh, the remaining comment has been addressed. Should I rebase/squash the commits before the merge can be done?

I'm not a maintainer of this repository, so I cannot answer that I'm afraid.

@dneto0 might be better positioned to decide (and merge this PR if it looks good).

**Comment by AGindinson** [comment]
@dneto0, could you please take a look?

**Pull request merged**
